docs/learn/learn-xcm-pallet.md

```diff
@@ -8,14 +8,119 @@ slug: ../learn-xcm-pallet
 ---
 
 The XCM pallet ([`pallet-xcm`](https://github.com/paritytech/polkadot/tree/master/xcm/pallet-xcm))
-provides a developer-friendly interface for most of the common XCM messages.
+provides a set of pre-defined, commonly used XCVM programs in the form of a set of extrinsics.
 
-This pallet provides default implementations for many traits required by `XcmConfig`, which is
-highly useful for configuration.
+This pallet provides some default implementations for traits required by `XcmConfig`. The XCM
+executor is also included as an associated type within the pallet's configuration.
 
-It defines a set of extrinsics that can be utilized to build XCVM programs, either to target the
-local or external chains. Pallet-xcm's functionality is separated into three categories:
+Where the XCM format defines a set of instructions used to construct XCVM programs, `pallet-xcm`
+defines a set of extrinsics that can be utilized to build XCVM programs, either to target the local
+or external chains. `pallet-xcm`'s functionality is separated into three categories:
 
-1. Primitive, dispatchable functions to locally execute an XCM message.
+:::note
+
+Remember, all XCMs are XCVM programs that follow
+[the XCM format](https://github.com/paritytech/xcm-format). It is the job of the XCM executor is to
+handle and execute these programs.
+
+:::
+
+1. Primitive, dispatchable functions to locally execute an XCM.
 2. High-level, dispatchable functions for asset transfers.
 3. Version negotiation-specific dispatchable functions.
+
+## Primitive Extrinsics
+
+There are two primary primitive extrinsics. These extrinsics handle sending and executing XCVM
+programs as dispatchable functions within the pallet.
+
+1. `execute` - This call contains direct access to the XCM executor. It is tje job of the executor
+   to check the message and ensure that no barrier/filter will block the execution of the XCM. Once
+   it is deemed valid, the message will then be _locally_ executed, therein returning the outcome as
+   an event. This operation is executed on behalf of whichever account has signed the extrinsic.
+   It's possible for only a partial execution to occur.
+2. `send` - This call specifies where a message should be sent externally to a particular
+   destination, i.e. a parachain, smart contract, or any system which is governed by consensus. As
+   before with `execute`, the Executor checks the origin, destination, and message and is then sent
```

Comment:
No? I don't think the local XCM executor is consulted before the XCM is sent.

Reply:
https://github.com/paritytech/polkadot/blob/5feae5d7e5630e9e3837a2f9517c8ae0a376dea4/xcm/pallet-xcm/src/lib.rs#LL769C5-L769C5

Reply:
A mistake on my part - only for local execution is the executor called. Send merely sends it _to be_ executed on the destination.